pkg/go.mod

```diff
@@ -14,23 +14,23 @@ require (
 	github.com/stretchr/testify v1.10.0
 	github.com/testcontainers/testcontainers-go/modules/k3s v0.32.0
 	github.com/wk8/go-ordered-map/v2 v2.1.8
-	golang.org/x/mod v0.22.0
-	golang.org/x/tools v0.29.0
-	helm.sh/helm/v3 v3.14.4
-	k8s.io/api v0.30.3
-	k8s.io/apimachinery v0.30.3
-	k8s.io/client-go v0.30.3
-	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
+	golang.org/x/mod v0.24.0
+	golang.org/x/tools v0.33.0
+	helm.sh/helm/v3 v3.17.3
+	k8s.io/api v0.33.0
```

Comment:
I would prefer if we backed out these changes to the go version and the package bumps. I'm not opposed to upgrading, but I'd prefer if it happened a bit more incrementally as it also looks like it introduces some CRD changes to the core structs we're importing some places.

Reply:
IIRC these come from the bump of helm. With the k8s SDK, we don't get too much say so in how this happen :(

FWIW it is in a distinct commit: https://github.com/redpanda-data/redpanda-operator/pull/824/commits/d0f56dfeb2370536f3087f75a5d91928e1367c9d (Nice Rafal!)

Reply:
I don't see other option as to leave this dependency bumped. If you have any idea how to bump rpk code without bumping k8s.io deps I would like to learn.